tests/unit/restapi/test_db_models.py

```diff
@@ -907,3 +911,57 @@ def test_db_invalid_resource_dependency_fails(
         except IntegrityError as err:
             db.session.rollback()
             raise err
+
+
+def test_task_engine_yaml_creation_works(
+    db: SQLAlchemy,
+    registered_plugin_id: int,
+    registered_entry_point_id: int,
+) -> None:
+    entrypoint = viewsdb.get_latest_entry_point(
+        db, resource_id=registered_entry_point_id
+    )
+    plugin = viewsdb.get_latest_plugin(db, resource_id=registered_plugin_id)
+    files = viewsdb.get_latest_plugin_files(db, plugin_resource_id=registered_plugin_id)
+
+    types = {
+        param.parameter_type.name: param.parameter_type.structure
+        for file in files
+        for task in file.tasks
+        for param in task.input_parameters + task.output_parameters
+        if param.parameter_type.name not in BUILTIN_TYPES
+    }
+
+    parameters = {param.name: param.default_value for param in entrypoint.parameters}
+
+    tasks = {
+        task.plugin_task_name: {
+            "plugin": ".".join(
+                [plugin.name, Path(file.filename).stem, task.plugin_task_name]
+            ),
+            "inputs": [
+                {"name": param.name, "type": param.parameter_type.name}
+                for param in task.input_parameters
+            ],
+            "outputs": [
+                {param.name: param.parameter_type.name}
+                for param in task.output_parameters
+            ],
+        }
+        for file in files
+        for task in file.tasks
+    }
+
+    graph = yaml.safe_load(entrypoint.task_graph)["graph"]
```

Comment:
I know in the earlier test I included an explicit `graph:` section in the entrypoint, although I feel like we can omit that to save a level of indentation.

Reply:
agreed. Since the user will only have to provide the graph section, it doesn't make sense for them to nest everything. I just added that since it was in the example registered entrypoint.